### PR TITLE
PHPCS fixes for 1.9.1-dev

### PR DESCRIPTION
**Pull request by paulschreiber**

### CODE REVIEW

**Comment by paulschreiber** [comment]
Two lines generate escaping warnings:
* class-wpcom-liveblog-lazyloader.php:142
* class-wpcom-liveblog-amp-template.php:97

Please inspect them and choose an appropriate escaping function, or add a `phpcs:ignore` comment.

File: composer.json [hunk 1]
```diff
@@ -5,6 +5,9 @@
     "type"       : "wordpress-plugin",
     "license"    : "GPL-2.0+",
     "require"    : {
+        "wp-coding-standards/wpcs": "^1.0.0",
```
Comment (GaryJones) [inline composer.json]: These should be `require-dev` dependencies.

File: phpcs.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
```
Comment (GaryJones) [inline phpcs.xml]: This should be named `.phpcs.xml.dist` or `phpcs.xml.dist`.

File: phpcs.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+<ruleset name="WordPress Coding Standards for Liveblog">
```
Comment (GaryJones) [inline phpcs.xml]: Name should be `"Custom ruleset for Liveblog"`. It's a ruleset, not a standard.

File: phpcs.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+	<rule ref="WordPress-VIP" />
+	<rule ref="WordPress-Extra" />
```
Comment (GaryJones) [inline phpcs.xml]: There should be no spaces before closing `/>`. This is XML, not XHTML.

File: phpcs.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+	<arg value="s"/>
```
Comment (GaryJones) [inline phpcs.xml]: Include the progress (`p`), colors, extensions, basepath and parallel args - this plugin is PHP 5.6 and above, so PHPCS 3.x will be installed (3.3.1 currently).

File: phpcs.xml (hunk 2 above)
Comment (paulschreiber) [reply phpcs.xml]: I based this on the work done at https://github.com/Automattic/amp-wp/ cc @westonruter

File: phpcs.xml (hunk 3 above)
Comment (paulschreiber) [reply phpcs.xml]: Also copied from amp-wp.

File: phpcs.xml (hunk 4 above)
Comment (paulschreiber) [reply phpcs.xml]: Also copied from amp-wp.

File: composer.json (hunk 1 above)
Comment (paulschreiber) [reply composer.json]: fixed with 268daa4.

File: phpcs.xml (hunk 4 above)
Comment (paulschreiber) [reply phpcs.xml]: Fixed with 0ca4449.

File: phpcs.xml (hunk 5 above)
Comment (paulschreiber) [reply phpcs.xml]: I was previously told not to use `--colors`:
https://github.com/WordPress/gutenberg/pull/1685#issuecomment-312780011

Comment (paulschreiber) [reply phpcs.xml]: `extensions` is already present.

Comment (paulschreiber) [reply phpcs.xml]: What do you want for `basepath` and `parallel`? @westonruter any thoughts here?

File: phpcs.xml (hunk 2 above)
Comment (GaryJones) [reply phpcs.xml]: @westonruter should know better :-)

I've created https://github.com/Automattic/amp-wp/issues/1324 for them to update it at AMP.

File: phpcs.xml (hunk 3 above)
Comment (GaryJones) [reply phpcs.xml]: This is just about being accurate and educating developers.

For your reference (since I see you on many different repos doing PRs improving PHPCS setups like this - and thank you for that!), here's my [plugin boilerplate `.phpcs.xml.dist`](https://github.com/GaryJones/plugin-boilerplate/blob/master/.phpcs.xml.dist) file, which applies many good practices.

File: phpcs.xml (hunk 2 above)
Comment (paulschreiber) [reply phpcs.xml]: Renamed with 5767fb0.

File: phpcs.xml (hunk 3 above)
Comment (paulschreiber) [reply phpcs.xml]: Fixed with 14a4a9e.

File: phpcs.xml (hunk 5 above)
Comment (paulschreiber) [reply phpcs.xml]: Fixed based on your boilerplate. 15df6b2.

Comment (GaryJones) [reply phpcs.xml]: `basepath="."` is enough to remove everything before the current directory.
`parallel="8"` is a good default, though I put it to 50 on my plugins - I've not yet encountered a drawback, though that doesn't mean there isn't one.

File: phpcs.xml.dist [hunk 6]
```diff
@@ -1,8 +1,9 @@
 <?xml version="1.0"?>
 <ruleset name="Custom ruleset for Liveblog">
 
-	<rule ref="WordPress-VIP"/>
+	<rule ref="WordPress-Core"/>
```
Comment (GaryJones) [inline phpcs.xml.dist]: No need to include `WordPress-Core` if including `WordPress-Extra`, as Extra already includes Core.

**Comment by paulschreiber** [comment]
Updated to use `WordPress` as the ruleset.

**Comment by paulschreiber** [comment]
Failure is a Travis error:
```
The command "bash bin/install-wp-tests.sh wordpress_test root '' localhost $WP_VERSION" failed and exited with 7 during .
```

**Comment by paulschreiber** [comment]
First, you complained about the spacing in the XML tags. You did not mention anything about the ruleset. Then, I updated the rulesets to Core+Extra+VIPMinimum. You said to remove Core. Then, I changed the rulesets to WordPress+VIPMinimum, and you said not to use WordPress.

This was a ridiculous amount of back-and-forth for two lines of XML.

It would have been better if you had said up front:
> Use the `WordPress-Extra` and `WordPressVIPMinimum` rulesets. Do not use `WordPress` or `WordPress-Core`. You may use `WordPress-Docs` if you want.

or…accepted the pull request and made the change yourself.

**Comment by paulschreiber** [comment]
@philipjohn I have updated this for the released version of 1.9. I caught many more errors. Please run `phpcs` on the code before releasing.

**Comment by GaryJones** [comment]
@paulschreiber 	Thanks for the new commits.

Within the File Changes, I see a whole load of bits that aren't related to CS - changes in version numbers and so on.

Could you please try rebasing on to `master` again?

If that works it should greatly reduce the amount of code to review, and the number of commits.

Thanks!

File: liveblog.php [hunk 7]
```diff
@@ -838,7 +838,7 @@ public static function get_entries_paged( $page, $last_known_entry = false, $id
 			//If no page is passed but entry id is, we search for the correct page.
 			if ( false === $page && false !== $id ) {
 				$index = array_search( (int) $id, array_keys( $entries ), true );
-				$index = $index++;
+				$index++;
```
Comment (GaryJones) [inline liveblog.php]: While the change probably makes sense, I don't think this is something that PHPCS picks up as a violation, is it? (Happy to be shown I'm wrong!)

Comment (paulschreiber) [reply liveblog.php]: PHPCS flagged `$index = $index + 1`. I half-fixed it. Was finishing the fix.

**Comment by paulschreiber** [comment]
I merged master back in. Viewing the changes without whitespace-only diffs looks good:
https://github.com/Automattic/liveblog/pull/486/files?w=1

File: classes/class-wpcom-liveblog-wp-cli.php [hunk 8]
```diff
@@ -59,15 +59,17 @@ public function fix_archive( $args, $assoc_args ) {
 			$edit_entries  = $entries_query->get_all_edits( array( 'post_id' => $post_id ) );
 
 			// find correct comment_ids to replace incorrect meta_values
-			$correct_ids_array = $wpdb->get_results(
+			$correct_ids_array = $wpdb->get_results( // phpcs:ignore
```
Comment (GaryJones) [inline classes/class-wpcom-liveblog-wp-cli.php]: Can you list all of the checks being ignored, rather than a blanket ignore?

File: classes/class-wpcom-liveblog-wp-cli.php [hunk 9]
```diff
@@ -101,9 +103,10 @@ public function fix_archive( $args, $assoc_args ) {
 
 								// If this isnt a dry run we can run the database Update.
 								if ( false === $is_dryrun ) {
-									$wpdb->update(
+									$wpdb->update( // phpcs:ignore
 										$wpdb->commentmeta,
 										array( 'meta_value' => $correct_ids[ $i ] ),
+                                        // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
```
Comment (GaryJones) [inline classes/class-wpcom-liveblog-wp-cli.php]: This line has space indentation, not tabs.

File: classes/class-wpcom-liveblog-wp-cli.php [hunk 10]
```diff
@@ -101,9 +103,10 @@ public function fix_archive( $args, $assoc_args ) {
 
 								// If this isnt a dry run we can run the database Update.
 								if ( false === $is_dryrun ) {
-									$wpdb->update(
+									$wpdb->update( // phpcs:ignore
```
Comment (GaryJones) [inline classes/class-wpcom-liveblog-wp-cli.php]: List all of the checks being ignored.

File: classes/class-wpcom-liveblog-wp-cli.php [hunk 11]
```diff
@@ -59,15 +59,17 @@ public function fix_archive( $args, $assoc_args ) {
 			$edit_entries  = $entries_query->get_all_edits( array( 'post_id' => $post_id ) );
 
 			// find correct comment_ids to replace incorrect meta_values
-			$correct_ids_array = $wpdb->get_results(
+			$correct_ids_array = $wpdb->get_results( // phpcs:ignore
 				$wpdb->prepare(
 					"SELECT comment_id FROM $wpdb->comments
 				 WHERE comment_post_id = %d AND comment_id NOT IN
 				 ( SELECT $wpdb->commentmeta.comment_id FROM $wpdb->commentmeta
```
Comment (GaryJones) [inline classes/class-wpcom-liveblog-wp-cli.php]: This line, and the next four lines, have some mixed tabs + space indentations.

File: classes/class-wpcom-liveblog-wp-cli.php [hunk 12]
```diff
@@ -114,26 +117,28 @@ public function fix_archive( $args, $assoc_args ) {
 			}
 
 			// find comment_ids object with correct content for replacement
-			$correct_contents = $wpdb->get_results(
+			$correct_contents = $wpdb->get_results( // phpcs:ignore
```
Comment (GaryJones) [inline classes/class-wpcom-liveblog-wp-cli.php]: List all of the checks being ignored.

File: classes/class-wpcom-liveblog-wp-cli.php [hunk 13]
```diff
@@ -114,26 +117,28 @@ public function fix_archive( $args, $assoc_args ) {
...
 					 HAVING count(comment_content) = 2
-					 ORDER BY comment_id ASC", $post_id
+					 ORDER BY comment_id ASC",
+					$post_id
 				)
 			);
 
 			// find comment_ids that NEED to be replaced
-			$entries_replace = $wpdb->get_results(
+			$entries_replace = $wpdb->get_results( // phpcs:ignore
```
Comment (GaryJones) [inline classes/class-wpcom-liveblog-wp-cli.php]: List all of the checks being ignored.

File: classes/class-wpcom-liveblog-wp-cli.php [hunk 14]
```diff
@@ -151,7 +156,7 @@ public function fix_archive( $args, $assoc_args ) {
 					$content = $correct_contents[ $replaced ]->comment_content;
 
 					if ( false === $is_dryrun ) {
-						$wpdb->update(
+						$wpdb->update( // phpcs:ignore
```
Comment (GaryJones) [inline classes/class-wpcom-liveblog-wp-cli.php]: List all of the checks being ignored.

File: composer.json [hunk 15]
```diff
@@ -4,7 +4,10 @@
     "homepage"   : "https://github.com/Automattic/liveblog",
```
Comment (GaryJones) [inline composer.json]: In terms of code standards, this whole file should be tab-indented.

File: liveblog.php [hunk 16]
```diff
@@ -1625,11 +1630,11 @@ private static function status_header_with_message( $status, $message ) {
 
 			$status                       = absint( $status );
 			$official_message             = isset( $wp_header_to_desc[ $status ] ) ? $wp_header_to_desc[ $status ] : '';
-			$wp_header_to_desc[ $status ] = self::sanitize_http_header( $message );
+			$wp_header_to_desc[ $status ] = self::sanitize_http_header( $message ); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.OverrideProhibited
```
Comment (GaryJones) [inline liveblog.php]: I've never even seen or heard of this global before! TIL.

File: phpcs.xml.dist [hunk 17]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<ruleset name="Custom ruleset for Liveblog">
+
+	<rule ref="WordPress-Extra"/>
```
Comment (GaryJones) [inline phpcs.xml.dist]: Also add `WordPress-Docs`? It's not fully complete, but it's a good start for what it does cover. Could leave for a separate PR.

File: phpcs.xml.dist [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+	<rule ref="WordPressVIPMinimum"/>
+
+	<arg value="sp"/>
+	<arg name="colors"/>
+	<arg name="basepath" value="."/>
+	<arg name="parallel" value="50"/>
+	<arg name="extensions" value="php"/>
+	<file>.</file>
+
```
Comment (GaryJones) [inline phpcs.xml.dist]: Would like to see text domain checks enabled, and prefixes, and some link references.

See https://github.com/GaryJones/plugin-boilerplate/blob/master/.phpcs.xml.dist, and consider copying this, and amending it with the liveblog bits (no need to include Neutron).

PHPCompatibilityWP standard would also then need to be added to `composer.json` and Travis config.

File: phpcs.xml.dist (hunk 17 above)
Comment (philipjohn) [reply phpcs.xml.dist]: Yep, let's do a separate issue outside  of this PR :+1:

File: phpcs.xml.dist (hunk 18 above)
Comment (philipjohn) [reply phpcs.xml.dist]: Let's leave this for a separate change

**Comment by paulschreiber** [comment]
@philipjohn I've addressed the issues @GaryJones flagged.
